kernel/src/net/socket/netlink/route/kernel/link.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Handle link-related requests.
+
+use core::num::{NonZero, NonZeroU32};
+
+use aster_bigtcp::iface::InterfaceType;
+
+use crate::{
+    net::{
+        iface::{Iface, IFACES},
+        socket::netlink::route::message::{
+            attr::link::LinkAttr, CMessageSegmentHeader, CSegmentType, GetRequestFlags,
+            LinkSegment, LinkSegmentBody, MsgSegment, SegmentHeaderCommonFlags,
+        },
+    },
+    prelude::*,
+    util::net::CSocketAddrFamily,
+};
+
+pub fn do_get_link(request_segment: &LinkSegment) -> Result<Vec<MsgSegment>> {
+    validate_getlink_request_body(request_segment.body())?;
+
+    let dump_all = {
+        let flags = GetRequestFlags::from_bits_truncate(request_segment.header().flags);
+        flags.contains(GetRequestFlags::DUMP)
+    };
+    let required_index = request_segment.body().index.map(NonZeroU32::get);
+    let required_name = request_segment.attrs().iter().find_map(|attr| {
+        if let LinkAttr::Name(name) = attr {
+            Some(name.to_str().unwrap())
+        } else {
+            None
+        }
+    });
+
+    if dump_all && required_index.is_some() {
+        return_errno_with_message!(
+            Errno::EINVAL,
+            "filtering by device index is not supported for link dumps"
```

Comment:
Sounds like it's not valid instead of not supported?
```suggestion
            "filtering by interface index is not valid for link dumps"
```